### PR TITLE
upstream gh action updates

### PR DESCRIPTION
**Pull request by AjayJagan**
Changing the gh actions as it is done in other artemiscloud repositories.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout the repo
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
```
Comment (ajaypratap003) [inline .github/workflows/ci.yml:50]:
```suggestion
     - name: Set up Node.js
        uses: actions/setup-node@v3
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      - name: Get node_modules cache
+        uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: |
+              ~/.yarn
+              **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+
+      - run: yarn install --frozen-lockfile
```
Comment (ajaypratap003) [inline .github/workflows/ci.yml:68]:
```suggestion
     -  name: Install dependencies
        if: steps.node-cache.outputs.cache-hit != 'true'
        shell: bash
        run: yarn install --frozen-lockfile
```

@AjayJagan Let's keep existing separate cache file i.e. `actions/cache-restore/action.yml` and modify if any changes required.  There are some check missing here for cache like  if: steps.node-cache.outputs.cache-hit != 'true'.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      #     yarn run build-storybook -- --quiet --loglevel silent
+
+      # - name: Deploy story book to github pages 🚀
+      #   uses: JamesIves/github-pages-deploy-action@v4.4.0
+      #   with:
+      #     branch: gh-pages # The branch the action should deploy to.
+      #     folder: storybook-static # The folder the action should deploy.
+
+      - name: Build the docker image
+        if: ${{ github.event_name == 'push' }}
```
Comment (ajaypratap003) [inline .github/workflows/ci.yml:89]:
```suggestion

```
We don't need this check as we need to make sure docker image build is working fine on PR. Below push check is fine for push the image.

**Review by ajaypratap003 — CHANGES_REQUESTED** [review]
Added few inline suggestions/comments. 
I noticed that build  docker image job is not running due to `push` check which is not required. 
Rest looks good to me.

**Comment by ajaypratap003** [comment]
@brusdev @RoddieKieley Please have a look into this PR whenever you guys have time.

**Review by ajaypratap003 — APPROVED** [review]
LGTM

**Review by brusdev — APPROVED** [review]
@AjayJagan nice work!!!

Could you squash your commits before merging this PR?

**Comment by AjayJagan** [comment]
> @AjayJagan nice work!!!
> 
> Could you squash your commits before merging this PR?

Thanks a lot! and yes we will sqash before merge(owing to a lot of commits 😅)

**Pull request merged**